src/main/java/com/google/sps/data/AvailabilityTimeSlot.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.auto.value.AutoValue;
+
+/**
+ * An AvailabilityTimeSlot represents a 15 minute chunk of time that has a date and time that are
+ * expressed as readable date and time fields as well as a more all-encompassing utcEncoding. It
+ * also has a boolean that tells whether or not the time slot has been selected.
+ */
+@AutoValue
+public abstract class AvailabilityTimeSlot {
+  public abstract String utcEncoding();
+
+  public abstract String time();
+
+  public abstract String date();
+
+  public abstract boolean selected();
+
+  public static AvailabilityTimeSlot create(
+      String utcEncoding, String time, String date, boolean selected) {
+    return builder()
+        .setUtcEncoding(utcEncoding)
+        .setTime(time)
+        .setDate(date)
+        .setSelected(selected)
+        .build();
+  }
+
+  static Builder builder() {
```

Comment:
This class is only used within this package, we have a separate class that is Availability that is more widely used and interacts with DAOs.

Reply:
In that case, perhaps everything here should be package private.

Reply:
I made the create function package-private and visible for testing. Leaving the fields (utcEncoding, etc.) public because they are accessed in the availabilityTable.jsp file.

Reply:
ack, marking resolved.